### DISCUSSION TITLE
Inverse for N x N matrix

### DISCUSSION THREAD

**Discussion by nithinsomu** [NVIDIA/warp#77] [General]
Is there a way to perform inversion of square matrices of any size within warp?

**Comment by mmacklin** [comment]
Hi @nithinsomu, currently we only provide `wp.inverse()` for mat22, mat33, and mat44 types.

There is an experimental Cholesky decomposition (and back solve) availble as `wp.dense_chol()`, but it's not well supported.

What size matrices do you need? And do you need an inverse or just a linear solve?

**Comment by nithinsomu** [comment]
Hi @mmacklin, Thanks for the response. Linear solve would be sufficient. Can the dense_chol handle matrices of size greater than 4 x 4? In my case I would need to do linear solve for matrices that have sizes more than 100 x 100.
